File: public/boilerplate/koa/src/utils/sign.js
import crypto from "crypto";
import config from "../config";

/**
 * 加密用户密码
 * @param {string} username 用户名
 * @param {string} password 密码
 */
const signPwd = (username, password) => {
  const { PWD_SALT } = config;
  const sign = crypto
    .createHash("md5")
    .update(username)
    .update(password)
    .update(password)
    .update(PWD_SALT)
    .update(PWD_SALT)
    .update(PWD_SALT)
    .digest("hex");
  return sign;
};

export default {
  signPwd,
};

File: public/boilerplate/gin/utils/sign.go
package utils

import (
	"crypto/md5"
	"encoding/hex"

	"github.com/fishjar/gin-rest-boilerplate/config"
)

// MD5Pwd 密码哈希函数
func MD5Pwd(username string, password string) string {
	salt := config.PWDSalt
	m := md5.New()
	m.Write([]byte(username))
	m.Write([]byte(password))
	m.Write([]byte(password))
	m.Write([]byte(salt))
	m.Write([]byte(salt))
	m.Write([]byte(salt))
	return hex.EncodeToString(m.Sum(nil))
}
